faunadb-java/src/test/java/com/faunadb/client/ClientSpec.java

```diff
@@ -2062,6 +2062,37 @@ public void shouldTestHasIdentity() throws Exception {
     }
   }
 
+  @Test
+  public void shouldTestHasCurrentToken() throws Exception {
+    Value createdInstance = serverClient.query(
+            Create(onARandomCollection(),
+                    Obj("credentials",
+                            Obj("password", Value("sekret"))))
+    ).get();
+
+    Value auth = serverClient.query(
+            Login(
+                    createdInstance.get(REF_FIELD),
+                    Obj("password", Value("sekret")))
+    ).get();
+
+    String secret = auth.get(SECRET_FIELD);
+
+    try (FaunaClient sessionClient = serverClient.newSessionClient(secret)) {
+      assertThat(
+              sessionClient.query(HasCurrentToken()).get().to(BOOLEAN).get(),
+              equalTo(true)
+      );
+    }
+
+    try (FaunaClient sessionClient = serverClient) {
```

Comment:
According to [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html), the try-with-resources statement is supposed to close the `FaunaClient` once the block is executed... if the `serverClient` is shared among the other test cases, why are the rest of the tests not failing? 🤔

Reply:
That's a good question, I will remove it in that case as it does not make sense.

Reply:
It seems to work fine as is, so I'd we keep like this since other test cases are following this pattern (at least until we figure out the correct way to do it!)